modules.d.ts

```diff
@@ -10,35 +10,247 @@ export declare const decodePresenceMessage: Types.PresenceMessageStatic['fromEnc
+ * ```
+ *
+ * If you do not provide this module, then attempting to access a channel’s {@link Types.RealtimeChannel.presence} property will cause a runtime error.
+ */
 export declare const RealtimePresence: unknown;
+
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to establish a connection with the Ably realtime service using a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) connection.
+ *
+ * To create a client that includes this module, include it in the `ModulesMap` that you pass to the {@link BaseRealtime.constructor}:
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { WebSocketTransport, FetchRequest });
+ * ```
+ *
+ * Note that network conditions, such as firewalls or proxies, might prevent the client from establishing a WebSocket connection. For this reason, you may wish to provide the `BaseRealtime` instance with the ability to alternatively establish a connection using a transport that is less susceptible to these external conditions. You do this by passing an alternative transport module, such as {@link XHRStreaming}, alongside `WebSocketTransport`:
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { WebSocketTransport, XHRStreaming, FetchRequest });
+ * ```
+ */
 export declare const WebSocketTransport: unknown;
+
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to establish a connection with the Ably realtime service using the browser’s [XMLHttpRequest API](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { XHRPolling, FetchRequest });
+ * ```
+ *
+ * Provide this module if you wish the client to have an alternative mechanism for connecting to Ably if it’s unable to establish a WebSocket connection.
+ */
 export declare const XHRPolling: unknown;
+
+// TODO how is it different to XHRPolling? How should the user choose between the two?
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to establish a connection with the Ably realtime service using the browser’s [XMLHttpRequest API](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { XHRPolling, FetchRequest });
+ * ```
+ *
+ * Provide this module if you wish the client to have an alternative mechanism for connecting to Ably if it’s unable to establish a WebSocket connection.
+ */
 export declare const XHRStreaming: unknown;
+
+// TODO what are the conditions in which somebody would do this? Is Fetch not always available in the browsers we’re targeting?
```

Comment:
Do you have any knowledge / thoughts that would help here, @owenpearson?

Reply:
Yeah `fetch` is supported by the evergreen browsers, but there are still all sorts of weird web-like js platforms out there. Let's not talk about any specific scenario where someone wants to avoid `fetch` but just list it as an alternative. Also, `XHRRequest` will of course still be needed to use the web comet transports.

Reply:
OK, I've left the wording as it is.